c/validation/adbc_validation.h

```diff
@@ -491,6 +491,14 @@ class StatementTest {
                                             const char* timezone);
 };
 
+template <typename CType>
+void StatementTest::TestSqlIngestType(ArrowType type,
```

Comment:
There's also `TestSqlIngestNumericType` and `TestSqlIngestTemporalType` that would be affected by the same issue, although these aren't actually used by any subclasses.

Maybe these should be private methods?

Reply:
Hmm if not used by subclasses we could keep them in the CC file to hopefully slim down compilation times a tiny bit (though header discipline etc is probably pretty bad anyways)

Reply:
> Hmm if not used by subclasses we could keep them in the CC file

Do you mean to make these free-standing functions in the module? I think the challenge there is that they ultimately wrap `TestSqlIngestType`, which relies on some of the class members

Reply:
I mean that private methods would be fine but ideally we can keep the body in the CC file and not in the header? Or is that not doable

Reply:
If you are talking about TestSqlIngestType we can move this back to the cc file if we explicitly instantiate the template types that subclasses can use there as well. 

The other methods already have their implementations in the cc module